rom/dev/src/flow/cold_reset/idev_id.rs

```diff
@@ -245,13 +245,13 @@ impl InitDevIdLayer {
         let sig = okmutref(&mut sig)?;
 
         // Verify the signature of the `To Be Signed` portion
-        let result = Crypto::ecdsa384_verify(env, &key_pair.pub_key, tbs.tbs(), sig)?;
-        if cfi_launder(result) == Ecc384Result::Success {
-            cfi_assert!(result == Ecc384Result::Success);
-        } else {
-            cfi_assert!(result != Ecc384Result::Success);
+        let mut verify_r = Crypto::ecdsa384_verify(env, &key_pair.pub_key, tbs.tbs(), sig)?;
+        if cfi_launder(verify_r != sig.r) {
```

Comment:
Does this actually do what you intend? I think the compiler is still allowed to optimize the input to cfi_launder; it's the output that makes a difference.

Perhaps this should be cfi_launder(verify_r) != sig.r?

Reply:
Yeah, this is no good: For example:

```
use std::hint::black_box;

pub fn func() -> bool {
    let val = 42;
    black_box(35 == val)
}
ASM:

playground::func:
	movb	$0, -1(%rsp)
	leaq	-1(%rsp), %rax
	#APP
	#NO_APP
	movzbl	-1(%rsp), %eax
	retq
```

As you can see, the compiler optimized out the comparison. However,

```
use std::hint::black_box;

pub fn func() -> bool {
    let val = 42;
    35 == black_box(val)
}
playground::func:
	movl	$42, -4(%rsp)
	leaq	-4(%rsp), %rax
	#APP
	#NO_APP
	cmpl	$35, -4(%rsp)
	sete	%al
	retq
```

Reply:
I verified this check with and without launder; there were one and two memcmps respectively; that said, I think your suggestion makes; I'll change it if I still see the desired assembly.

Reply:
Take a look at func5 here:

https://godbolt.org/z/8vMqvsKMn

Reply:
cfi_launder(verify_r) != sig.r does not work; the compiler is stripping the cfi_assert! in this case. The original check is correct; I can use references though.